### ISSUE TITLE
do you tranfer to classify on cifar10 dataset?

### ISSUE THREAD

**Issue by creat2012** [blaueck/tf-triplet-demo#7]
do you tranfer to classify on cifar10 dataset?

**Comment by blaueck** [comment]
Do you mean that transfer the model to classification task after being trained with triplet loss? I have not tried it. But it can be done by training a new classifier with image features extract by the model.

**Comment by creat2012** [comment]
> Do you mean that transfer the model to classification task after being trained with triplet loss? I have not tried it. But it can be done by training a new classifier with image features extract by the model.

yeah.
so, you mean, if i transfer the model to classification task, it will work? (in fact,  i try it , but it does not work. may be somewhere wrong . i don't know.)

**Comment by blaueck** [comment]
It should work. After trained with triplet loss, features from the same class should be closed to each other. How do you transfer to classification task?

**Comment by creat2012** [comment]
> It should work. After trained with triplet loss, features from the same class should be closed to each other. How do you transfer to classification task?

ok , i know. may be somewhere wrong. i will re-check it. thank you for your response.

**Comment by creat2012** [comment]
> It should work. After trained with triplet loss, features from the same class should be closed to each other. How do you transfer to classification task?

ClassificationAlexNet(
  (embedding_net): BaseAlexNet(
    (convnet): Sequential(
      (0): BatchNorm2d(3, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (1): Conv2d(3, 96, kernel_size=(11, 11), stride=(4, 4), padding=(2, 2))
      (2): BatchNorm2d(96, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (3): PReLU(num_parameters=1)
      (4): MaxPool2d(kernel_size=3, stride=2, padding=0, dilation=1, ceil_mode=False)
      (5): Conv2d(96, 256, kernel_size=(5, 5), stride=(1, 1), padding=(2, 2))
      (6): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (7): PReLU(num_parameters=1)
      (8): MaxPool2d(kernel_size=3, stride=2, padding=0, dilation=1, ceil_mode=False)
      (9): Conv2d(256, 384, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (10): BatchNorm2d(384, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (11): PReLU(num_parameters=1)
      (12): Conv2d(384, 384, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (13): BatchNorm2d(384, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (14): PReLU(num_parameters=1)
      (15): Conv2d(384, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (16): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (17): PReLU(num_parameters=1)
      (18): MaxPool2d(kernel_size=3, stride=2, padding=0, dilation=1, ceil_mode=False)
    )
    (fc): Sequential(
      (0): Linear(in_features=9216, out_features=4096, bias=True)
      (1): PReLU(num_parameters=1)
      (2): Dropout(p=0.5)
      (3): Linear(in_features=4096, out_features=4096, bias=True)
      (4): PReLU(num_parameters=1)
      (5): Dropout(p=0.5)
    )
  )
  (classifier): Linear(in_features=9216, out_features=10, bias=True)
)
freeze embedding_net.convnet.0.weight
freeze embedding_net.convnet.0.bias
freeze embedding_net.convnet.1.weight
freeze embedding_net.convnet.1.bias
freeze embedding_net.convnet.2.weight
freeze embedding_net.convnet.2.bias
freeze embedding_net.convnet.3.weight
freeze embedding_net.convnet.5.weight
freeze embedding_net.convnet.5.bias
freeze embedding_net.convnet.6.weight
freeze embedding_net.convnet.6.bias
freeze embedding_net.convnet.7.weight
freeze embedding_net.convnet.9.weight
freeze embedding_net.convnet.9.bias
freeze embedding_net.convnet.10.weight
freeze embedding_net.convnet.10.bias
freeze embedding_net.convnet.11.weight
freeze embedding_net.convnet.12.weight
freeze embedding_net.convnet.12.bias
freeze embedding_net.convnet.13.weight
freeze embedding_net.convnet.13.bias
freeze embedding_net.convnet.14.weight
freeze embedding_net.convnet.15.weight
freeze embedding_net.convnet.15.bias
freeze embedding_net.convnet.16.weight
freeze embedding_net.convnet.16.bias
freeze embedding_net.convnet.17.weight
embedding_net.fc.0.weight
embedding_net.fc.0.bias
embedding_net.fc.1.weight
embedding_net.fc.3.weight
embedding_net.fc.3.bias
embedding_net.fc.4.weight
classifier.weight
classifier.bias
====
as you saw,  i freeze the convnet's parameters(convnet's parameters from the training triplet loss model,  i initialize them by the pre trained model (triplet loss)), and just update the last serveral layers.
loss funtion is MSE. but the accuracy rate is about 55%
dataset : cifar10.
i did not user hard mining . (maybe , this is the key ? )

**Comment by blaueck** [comment]
55% of accuracy means model do have learned something, because it is much better than random. 
The low accuracy could be caused by the pool training of model with triplet loss. Without hard mining, it is very difficult or take very long time to converge with triplet loss.

**Comment by creat2012** [comment]
> 55% of accuracy means model do have learned something, because it is much better than random.
> The low accuracy could be caused by the pool training of model with triplet loss. Without hard mining, it is very difficult or take very long time to converge with triplet loss.

ok, it will try it again. thank you.
